### ISSUE TITLE
int main is malformed in all examples

### ISSUE THREAD

**Issue by soryy708** [jeremyong/SDLpp#4]
Isn't main supposed to take `int argc, char* argv[]` and explicitly `return`?

I know that standard C doesn't force that, but it seems like SDL does.

I have SDL2.0.1 and I haven't modified it. I can not compile any of the examples due to thrown errors:
1. `error C4716: 'SDL_main': must return a value` (can be fixed by returning 0 at the end of int main)
2. `error LNK2019: unresolved external symbol _SDL_main referenced in function _main` (can be fixed by making main take `int argc, char* argv[]`)

**Comment by jeremyong** [comment]
You're compiler isn't compliant to the standard. int main() returning nothing is a valid entry point for the C++ runtime.

**Comment by soryy708** [comment]
SDL_main.h declares main as follows: `extern C_LINKAGE int SDL_main(int argc, char *argv[]);` (line 89)

It also says:

```
/**
*  \file SDL_main.h
*
*  The application's main() function must be called with C linkage,
*  and should be declared like this:
*  \code
*  #ifdef __cplusplus
*  extern "C"
*  #endif
*  int main(int argc, char *argv[])
*  {
*  }
*  \endcode
*/
```

(from line 67 to line 80)

If they say so, then it seems like the correct way to use main is with `int argc, char *argv[]`.

---

And now about the return: My compiler is fine with me not having a return in a main function that has nothing to do with SDL / SDL_main. However, once I work with SDL / SDL_main, my compiler throws a fit.
Maybe it has something to do with the way SDL_main takes over the main function.

**Comment by soryy708** [comment]
["[...] because SDL defines a macro like this: `#define main SDL_main`
So the function you've written is actually called SDL_main and like any other function that is not the actual main function, if it doesn't return void, you have to give it a return statement."](http://stackoverflow.com/questions/15644883/sdl-main-must-return-a-value)

["On some platforms SDL defines main as SDL_main so that SDL can create their own main function that run code before or after your main function (which really is SDL_main). At least I think that is the reason. `#define main SDL_main`
This means that you have to write the return statement explicitly in main. I have never thought about this. I guess I will have to add a few return 0; in my own SDL projects."](http://www.cplusplus.com/forum/beginner/81759/)

So my compiler is fine. The main you see is not really _the_ main, but actually a faux by SDL.
The standard says that all non-void functions except _the_ main function must explicitly return. However, the main you see is actually SDL_main, a function that is not _the_ main but deceives you into thinking that it is.

TL;DR: You're wrong. While the standard says that main has the option of not returning explicitly, this is irrelevant because we're not talking about the same thing that the standard talks about.

**Comment by jeremyong** [comment]
Hmm you know what? I'd recommend using something else or just working off your own fork. I've been working on a framework that is much more low level (avoids the SDL drawing api) anyways and will not have as much time for this one.

> You're wrong. ...

You tend to take a very personal slant in all these discussions and GH issues you've been creating. Frankly speaking, I don't like it and I doubt many other project maintainers would either. This isn't a competition. It's not you vs them. It's just a collaboration and you should conduct yourself accordingly. Thanks.
